### ISSUE TITLE
Docsify with GitLab Pages Access Control Enabled Caused CORS

### ISSUE THREAD

**Issue by zjkyz8** [docsifyjs/docsify#1990]
#### Steps to reproduce
- SetUp a Gitlab Pages Project which use Docsify.
- Enable Gitlab Pages Access Control, follow this [Doc](https://docs.gitlab.com/ee/user/project/pages/pages_access_control.html)
- Access the doc page, you will ask to login gitlab
- Waiting a while(about 10 minute, make sure gitlab login token expire).
- Click a link to access other doc page.
- 

#### What is current behaviour
You will be redirected to 404 page, and you can see there are some CORS errors in dev tools console tab.  

<img width="1727" alt="image" src="https://user-images.githubusercontent.com/1856389/218236483-4a8b1283-b0ac-4338-b3dc-8f6f9674e034.png">

#### What is the expected behaviour
new doc page should be shown

#### My idea
When enable GitLab Pages Access Control, if token expired, Gitlab will redirect user browser to login page, this redirect is browser jump, not xhr access, so no CORS need to process. But for docsify, it use xhr to get doc from Gitlab, Gitlab still return redirect and expect browser to jump new page, so CORS caused.

#### Other relevant information

- [x] Bug does still occur when all/other plugins are disabled?

- Your OS: 
- Node.js version: 
- npm/yarn version: 
- Browser version: 
- Docsify version: 
- Docsify plugins: 

#### Please create a reproducible sandbox 

[![Edit 307qqv236](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/307qqv236)

**Comment by Koooooo-7** [comment]
Hi @zjkyz8 , the redirect event is from the backend, but docsify( xhr request) can not aware of it since it thought it request the `doc URL` by its origin url.  and the browser can do nothing here.

About the workaround, my assumption is trying to make the `session/token` not expired/ahead notice via a custom timer.

A. A timer to make the `heartbeat` to Gitlab, such as fetching the homepage.

B. A timer to get the new session/jumpage, once the session is expired, request the token and set the [requestHeaders](https://docsify.js.org/#/configuration?id=requestheaders) or just popup something show the login stuff.
```js
     const fetchHeaders = ()=>{
        // refresh the request headers
        return {
          // refreshed request headers with token/auth info
        };
      }
      setInterval(() => {
        window.$docsify.requestHeaders = fetchHeaders()
      }, 1000);
```

Not quite sure wether it works or not. May you have a try on it.

**Comment by zjkyz8** [comment]
@Koooooo-7 Sorry for reply late, like what you said, it seems that we can do nothing to fix this behavior based on xhr. 

Besides your solution, I found a new setting which name is auth_cookie_session_timeout, included by Gitlab 15.6, can make it better.

We can set this setting equal to 0 to make expired cookie become session cookie, if user does not close browser tab, auth never expire.

This is [doc](https://docs.gitlab.com/ee/administration/pages/) for ref, and I will close this issue.

**Comment by Koooooo-7** [comment]
Hi @zjkyz8 , no prob.
Glad to hear that it has been resolved.
